app/root.tsx

```diff
@@ -43,16 +56,39 @@ export const meta: MetaFunction = () => ({
 });
 
 type LoaderData = {
-    user: Awaited<ReturnType<typeof getUser>>;
+    user: User | null;
+    userSettings: UserSettings | null;
 };
 
 export const loader: LoaderFunction = async ({ request }) => {
+    const user = await getUser(request);
+    let userSettings: UserSettings | null = null;
+    if (user) {
+        userSettings = await getUserSettings(user.id);
+    }
+
     return json<LoaderData>({
-        user: await getUser(request)
+        user,
+        userSettings
     });
 };
 
 export default function Document() {
+    const submit = useSubmit();
+    const { userSettings } = useLoaderData<LoaderData>();
+    const [colorScheme, setColorScheme] = useState<ColorScheme>(
+        userSettings?.colorScheme ?? ColorScheme.Light
+    );
+    const toggleColorScheme = (value?: ColorScheme) =>
+        setColorScheme(
+            value || (colorScheme === ColorScheme.Dark ? ColorScheme.Light : ColorScheme.Dark)
+        );
+    const handleToggleColorScheme = () => {
+        submit(null, { method: 'post', action: '/settings/colorScheme', replace: true });
+        toggleColorScheme();
+    };
+    const isDarkTheme = colorScheme === ColorScheme.Dark;
```

Comment:
suggestion(non-blocking): It'd be pretty spiffy if we just had a custom hook that wrapped the `useMantineColorScheme` hook and returned nice boolean for us like `isDark` or `isLight`

Reply:
Love this suggestion! Seems like I've typed that same damn statement so many times, it would be great to have an easier way to get this information